app/text_buffer.py

```diff
@@ -732,15 +732,31 @@ def findCurrentPattern(self, direction):
       theRange = range(self.penRow)
     else:
       theRange = range(len(self.lines) - 1, self.penRow, -1)
-    for i in theRange:
-      found = localRe.search(self.lines[i])
+    if theRange:
```

Comment:
I cleaned up the code a little bit. It now repeatedly checks if found is ever not None and if not, it will keep trying.